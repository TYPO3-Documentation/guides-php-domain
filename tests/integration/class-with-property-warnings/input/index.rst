=============================================
PHP class with properties that cause warnings
=============================================

..  php:class:: GrumpyGoblin

    Welcome to the gloomy world of GrumpyGoblins, spreading grumbles and dissatisfaction!

    ..  php:property:: $hiddenSecret
        :type: string
        :public:
        :private:

        The grumpy goblin's hidden secret. Publicly private - an illegal combination!

    ..  php:property:: $ancientCurse
        :type: bool
        :static:
        :readonly:

        The ancient curse haunting all GrumpyGoblins. A static readonly curse - an illegal combination!

    ..  php:attr:: $complaintBox
        :type: string
        :public:

        The deprecated complaint box, once used by GrumpyGoblins to voice their discontent.
