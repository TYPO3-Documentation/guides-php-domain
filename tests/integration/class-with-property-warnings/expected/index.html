<!-- content start -->
    <div class="section" id="php-class-with-properties-that-cause-warnings">
    <h1>PHP class with properties that cause warnings</h1>

            <dl class="php class">
    <dt class="sig sig-object php"
        id="grumpygoblin">
        <em class="property"><span class="pre">class</span> </em>
        <span class="sig-name descname"><span class="pre">GrumpyGoblin</span></span>

    </dt>
    <dd>
        <p>Welcome to the gloomy world of GrumpyGoblins, spreading grumbles and dissatisfaction!</p><dl class="php property">
    <dt class="sig sig-object php" id="hiddensecret">
        <span class="sig-name modifier"><span class="pre">public</span></span>
 <span class="sig-name modifier"><span class="pre">private</span></span>
 <span class="pre">string</span> <span class="sig-name descname"><span class="pre">$hiddenSecret</span></span>

    </dt>
    <dd><p>The grumpy goblin&#039;s hidden secret. Publicly private - an illegal combination!</p></dd>
</dl>
<dl class="php property">
    <dt class="sig sig-object php" id="ancientcurse">
        <span class="sig-name modifier"><span class="pre">static</span></span>
 <span class="sig-name modifier"><span class="pre">readonly</span></span>
 <span class="pre">bool</span> <span class="sig-name descname"><span class="pre">$ancientCurse</span></span>

    </dt>
    <dd><p>The ancient curse haunting all GrumpyGoblins. A static readonly curse - an illegal combination!</p></dd>
</dl>
<dl class="php property">
    <dt class="sig sig-object php" id="complaintbox">
        <span class="sig-name modifier"><span class="pre">public</span></span>
 <span class="pre">string</span> <span class="sig-name descname"><span class="pre">$complaintBox</span></span>

    </dt>
    <dd><p>The deprecated complaint box, once used by GrumpyGoblins to voice their discontent.</p></dd>
</dl>

    </dd>
</dl>

    </div>

<!-- content end -->
